src/content/blog/astro-4100.mdx

```diff
@@ -0,0 +1,204 @@
+---
+title: "Astro 4.10"
+description: "Astro 4.10 is out with experimental type-safe environment variables."
+publishDate: "June 6, 2024"
+authors:
+  - matthew
+coverImage: "/src/content/blog/_images/astro-4100/header-astro-4.10.webp"
+socialImage: "/src/content/blog/_images/astro-4100/og-image-4.10.webp"
+lang: "en"
+---
+
+import BlogSummary from "/src/components/BlogSummary.astro";
+
+<BlogSummary>
+  Astro 4.10 is out with experimental type-safe environment variables
+</BlogSummary>
+
+Full release highlights include:
+
+- [**Experimental: astro:env module**](#experimental-astro-env)
+- [**Rewrite for all HTTP methods**](#rewrite-for-all-http-methods)
+- [**Embedding Astro in server frameworks**](#embedding-astro)
+- [**Container API helpers**](#container-api-helpers)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Experimental: astro:env
+
+Astro 4.10 introduces a new experimental built-in module, `astro:env`.
+
+Environment variables allow you to configure your app with different values in different environments. But this comes with a great deal of complexity:
+
+- Some variables are used in the client and some only on the server.
+- Server variables are often __secrets__, things like API keys that you absolutely do not want exposed in the client, and you also do not want inlined into the server build either (which can be viewed by anyone with access to the build output).
+- Some variables are __required__ for your app to function at all; whereas other are optional enhancements.
+- Runtimes such as Cloudflare and Deno have different APIs for reading variables, creating a dev/prod difference you need to deal with.
+
+We built `astro:env` to provide more control and structure over environment variables. Here's how you might define a schema for them in your config:
+
+```js
+import { defineConfig, envField } from 'astro/config';
+
+export default defineConfig({
+  experimental: {
+    env: {
+      schema: {
+        API_PORT: envField.number({
+          context: 'server',
+          access: 'secret',
+          default: 7000
+        }),
+        PUBLIC_DASHBOARD_V2: envField.boolean({
+          context: 'server',
+          access: 'public',
+          default: false
+        }),
+      }
+    }
+  }
+})
+```
+
+Once defined you can use your variables by importing them from the `astro:env/server` and `astro:env/client` modules like so:
+
+```js
+import { PUBLIC_DASHBOARD_V2, getSecret } from 'astro:env/server';
+
+if (PUBLIC_DASHBOARD_V2) {
+	const API_PORT = getSecret("API_PORT") // number
+	await fetch(`https://my-secret-api.com:${API_PORT}/v2`)
+}
+```
+
+Or in a client component, like so:
+
+```jsx
```

Comment:
My suggestion was not showing usage in jsx because, from what I saw with astro actions, many people were wondering if it was supposed to work in vue etc.

My suggestion is to
1. Specify explicitly it works no matter the ui fw, it's just JS!
2. Don't show usage with a specific ui framework but with an astro script (definitely debatable)